beacon-chain/p2p/pubsub.go

```diff
@@ -69,6 +69,8 @@ func (s *Service) PublishToTopic(ctx context.Context, topic string, data []byte,
 
 // SubscribeToTopic joins (if necessary) and subscribes to PubSub topic.
 func (s *Service) SubscribeToTopic(topic string, opts ...pubsub.SubOpt) (*pubsub.Subscription, error) {
+	s.awaitStateInitialized() // Genesis time and genesis validator root are required to subscribe.
```

Comment:
This isn't the correct place, this will end up blocking subscription of new topics.

Reply:
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/sync/service.go#L267

This will only call it after we receive the state initialized and synced events. 

https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/p2p/service.go#L377

This will end up being blocked here as no more state initialized events are emmited now.

Reply:
This solves the race condition and yes it should block subscription of new topics until the state is initialized.